src/main/java/com/fleotadezuta/youthprogrammanager/service/TicketTypeService.java

```diff
@@ -14,31 +17,46 @@ public class TicketTypeService {
     private final TicketTypeMapper ticketTypeMapper;
     private final TicketTypeRepository ticketTypeRepository;
 
-    public Flux<TicketTypeDto> getAllTicketTypes() {
+    public Flux<TicketTypeDto> getAllTicketTypes(UserDetails userDetails) {
+        if (!userDetails.getUserType().equals("ADMIN")) {
+            return Flux.error(new ResponseStatusException(HttpStatus.NOT_FOUND, "User not authorized to report participation"));
+        }
         return ticketTypeRepository.findAll()
                 .map(ticketTypeMapper::fromTicketTypeDocumentToTicketTypeDto);
     }
 
-    public Mono<TicketTypeDto> getTicketTypeById(String id) {
+    public Mono<TicketTypeDto> getTicketTypeById(UserDetails userDetails, String id) {
+        if (!userDetails.getUserType().equals("ADMIN")) {
+            return Mono.error(new ResponseStatusException(HttpStatus.NOT_FOUND, "User not authorized to report participation"));
+        }
         return ticketTypeRepository.findById(id)
                 .map(ticketTypeMapper::fromTicketTypeDocumentToTicketTypeDto);
     }
 
-    public Mono<TicketTypeDto> addTicketType(TicketTypeDto ticketTypeDto) {
+    public Mono<TicketTypeDto> addTicketType(UserDetails userDetails, TicketTypeDto ticketTypeDto) {
+        if (!userDetails.getUserType().equals("ADMIN")) {
+            return Mono.error(new ResponseStatusException(HttpStatus.NOT_FOUND, "User not authorized to report participation"));
+        }
         return Mono.just(ticketTypeDto)
                 .map(ticketTypeMapper::fromTicketTypeDtoToTicketTypeDocument)
                 .flatMap(ticketTypeRepository::save)
                 .map(ticketTypeMapper::fromTicketTypeDocumentToTicketTypeDto);
     }
 
-    public Mono<TicketTypeDto> deletedTicketType(String id) {
+    public Mono<TicketTypeDto> deletedTicketType(UserDetails userDetails, String id) {
+        if (!userDetails.getUserType().equals("ADMIN")) {
+            return Mono.error(new ResponseStatusException(HttpStatus.NOT_FOUND, "User not authorized to report participation"));
+        }
         return ticketTypeRepository.findById(id)
                 .flatMap(ticketType -> ticketTypeRepository.deleteById(id)
                         .then(Mono.just(ticketType)))
                 .map(ticketTypeMapper::fromTicketTypeDocumentToTicketTypeDto);
     }
 
-    public Mono<TicketTypeDto> updateTicketType(String id, TicketTypeDto ticketTypeDto) {
+    public Mono<TicketTypeDto> updateTicketType(UserDetails userDetails, String id, TicketTypeDto ticketTypeDto) {
+        if (!userDetails.getUserType().equals("ADMIN")) {
+            return Mono.error(new ResponseStatusException(HttpStatus.NOT_FOUND, "User not authorized to report participation"));
+        }
```

Comment:
Egyfelől az összes hibaüzenet esetében igaz, hogy inkonzisztens a status code-dal. Vagy ne áruljuk el, hogy létezik a jegy típus az üzenettel sem, vagy pedig a status is annak megfelelő legyen.

Másfelől az üzenet nem az adott interakciónak a megfelelő.

Harmadrészt service rétegből nem kellene ResponseStatusException-t dobni, mert ez feltételezi, hogy milyen környezetből (valamilyen controller) lesznek hívva a metódusok. Szerencsésebb lenne egy saját exception-t definiálni (`UnauthorizedException`), amit a controllerben kezelve továbbdobsz majd `ResponseStatusException`-ként